src/trie.rs

```diff
@@ -30,15 +34,35 @@ impl Trie {
 		Self(QTrie::new())
 	}
 
-	pub(crate) fn insert(&mut self, domain: &str) {
+	/// add domain to trie, and return true if the domain was already added by the **same** list.
```

Comment:
```suggestion
	/// Add a domain to this trie. If the domain is not already in the trie, it will
	/// be added with the list marked as its single source and returns true. If the
	/// domain is already in the trie but was not marked as coming from the list, it
	/// will be marked as coming from the list as well and returns true. If the
	/// domain is already in the trie and is marked as coming from the list, it will
	/// return false.
```